File: repos_estudiantes_ALVAREZ_primera_entrega/CHAMBI_BARRETO_CARLOS_ERNESTO/INTRODUCCION_A_POO/EJERCICIO13/Fruta.java
package INTRODUCCION_A_POO.EJERCICIO13;

public class Fruta {
    public String nombre, tipo;
    public int nroVitaminas;
    public String[] vitaminas;

    public Fruta(String nombre, String tipo, String[] vitaminas) {
        this.nombre = nombre;
        this.tipo = tipo;
        this.vitaminas = vitaminas;
        this.nroVitaminas=vitaminas.length;
    }

    public String getNombre() {
        return nombre;
    }

    public String getTipo() {
        return tipo;
    }

    public int getNroVitaminas() {
        return nroVitaminas;
    }

    public String[] getVitaminas() {
        return vitaminas;
    }

    public void mostrarVitaminas(){
        for(int i = 0; i<vitaminas.length; i++){
            System.out.print(vitaminas[i] + " " );
        }
        System.out.println(" ");
    }

}

File: repos_estudiantes_ALVAREZ_primera_entrega/APAZA_PACHECO_BRYAN_ORLANDO/POO/Ejer13/Fruta.py
# a)
class Fruta:
    def __init__(self,nombre,tipo,nroVitaminas,v):
        self.nombre=nombre
        self.tipo=tipo
        self.nroVitaminas=nroVitaminas
        self.v=v  

    def getNombre(self):
        return self.nombre
    def setNombre(self,nombre):
        self.nombre=nombre

    def getTipo(self):
        return self.tipo
    def setTipo(self,tipo):
        self.tipo=tipo

    def getNroVitaminas(self):
        return self.nroVitaminas
    def setNroVitaminas(self,nro):
        self.nroVitaminas=nro

    def getV(self):
        return self.v
    def setV(self,v):
        self.v=v

f1=Fruta("Kiwi","Subtropical",3,["C","E","K"])
f2=Fruta("Naranja","Cítrica",4,["C","A","B1","B2"])
f3=Fruta("Fresa","Templada",2,["C","B9"])

# b)
frutas=[f1,f2,f3]
fruta_mas_vitaminas=max(frutas,key=lambda f:f.nroVitaminas)
print("Fruta con más vitaminas:",fruta_mas_vitaminas.nombre)

# c) 
print("Vitaminas de Kiwi:",f1.v)

# d) Cuantas v ejemplo C
total_citricas=sum(v.count("C") for f in frutas for v in f.v)
print("Cantidad de vitaminas cítricas:",total_citricas)

# e) Ordenar segun vitaminas xd
for f in frutas:
    f.v.sort()
for f in frutas:
    print(f.nombre,":",f.v)
